src/hotspot/share/gc/g1/g1SegmentedArray.hpp

```diff
@@ -29,147 +29,147 @@
   // check only afterwards if the allocation has been successful).
   uint volatile _next_allocate;
 
 public:
-  G1SegmentedArrayBuffer(uint elem_size, uint num_elems, G1SegmentedArrayBuffer* next);
-  ~G1SegmentedArrayBuffer();
+  G1SegmentedArraySegment(uint slot_size, uint num_slots, G1SegmentedArraySegment* next);
+  ~G1SegmentedArraySegment();
 
-  G1SegmentedArrayBuffer* volatile* next_addr() { return &_next; }
+  G1SegmentedArraySegment* volatile* next_addr() { return &_next; }
 
-  void* get_new_buffer_elem();
+  void* get_new_slot();
 
-  uint num_elems() const { return _num_elems; }
+  uint num_slots() const { return _num_slots; }
 
-  G1SegmentedArrayBuffer* next() const { return _next; }
+  G1SegmentedArraySegment* next() const { return _next; }
 
-  void set_next(G1SegmentedArrayBuffer* next) {
+  void set_next(G1SegmentedArraySegment* next) {
     assert(next != this, " loop condition");
     _next = next;
   }
 
-  void reset(G1SegmentedArrayBuffer* next) {
+  void reset(G1SegmentedArraySegment* next) {
     _next_allocate = 0;
     assert(next != this, " loop condition");
     set_next(next);
-    memset((void*)_buffer, 0, (size_t)_num_elems * _elem_size);
+    memset((void*)_segment, 0, (size_t)_num_slots * _slot_size);
   }
 
-  uint elem_size() const { return _elem_size; }
+  uint slot_size() const { return _slot_size; }
 
-  size_t mem_size() const { return sizeof(*this) + (size_t)_num_elems * _elem_size; }
+  size_t mem_size() const { return sizeof(*this) + (size_t)_num_slots * _slot_size; }
 
   uint length() const {
-    // _next_allocate might grow larger than _num_elems in multi-thread environments
+    // _next_allocate might grow larger than _num_slots in multi-thread environments
     // due to races.
-    return MIN2(_next_allocate, _num_elems);
+    return MIN2(_next_allocate, _num_slots);
   }
 
-  // Copies the (valid) contents of this buffer into the destination.
+  // Copies the (valid) contents of this segment into the destination.
   void copy_to(void* dest) const {
-    ::memcpy(dest, _buffer, length() * _elem_size);
+    ::memcpy(dest, _segment, length() * _slot_size);
   }
 
-  bool is_full() const { return _next_allocate >= _num_elems; }
+  bool is_full() const { return _next_allocate >= _num_slots; }
 };
 
-// Set of (free) G1SegmentedArrayBuffers. The assumed usage is that allocation
-// to it and removal of elements is strictly separate, but every action may be
+// Set of (free) G1SegmentedArraySegments. The assumed usage is that allocation
+// to it and removal of segments is strictly separate, but every action may be
 // performed by multiple threads at the same time.
 // Counts and memory usage are current on a best-effort basis if accessed concurrently.
 template<MEMFLAGS flag>
-class G1SegmentedArrayBufferList {
-  static G1SegmentedArrayBuffer<flag>* volatile* next_ptr(G1SegmentedArrayBuffer<flag>& node) {
-    return node.next_addr();
+class G1SegmentedArrayFreeList {
```

Comment:
Thanks Albert.
In fact I intended to change it "G1SegmentedArrayFreeList", I think this is more readable, and G1SegmentedArraySegmentList is a bit redundant.
Please kindly let me know if you think otherwise.

Reply:
There's a class named `G1SegmentedArraySegment`, so having `G1SegmentedArraySegmentList` as a list of `G1SegmentedArraySegment` makes perfect sense and is more consistent, IMO. Ofc, this is quite subjective; either is fine to me.

Reply:
Thanks Albert, I'll keep "G1SegmentedArrayFreeList". :)